os/include/sys/socket.h

```diff
@@ -303,8 +303,10 @@ int getsockname(int sockfd, FAR struct sockaddr *addr, FAR socklen_t *addrlen);
 */
 int getpeername(int s, struct sockaddr *name, socklen_t *namelen);
 
+#ifdef CONFIG_NET_SOCKET
 ssize_t recvmsg(int sockfd, struct msghdr *msg, int flags);
 ssize_t sendmsg(int sockfd, struct msghdr *msg, int flags);
```

Comment:
CONFIG_NET_SOCKET depends on CONFIG_NET_LWIP. That means source files in os/net/socket are dependent on LWIP. On the other hand, prototypes defined in this include/sys/socket.h provide general function formats regardless of the type of network stack, I believe. My point is that in case recvmsg and sendmsg are dependent on LWIP, their prototype should be defined in os/net/socket.h not here. However, if recvmsg and sendmsg are also general types of function, then it should not be dependend on CONFIG_NET_SOCKET. Please let me know your opinion. Thank you!
In my opinion, those two functions recvmsg and sendmsg should be moved into socket.h in

Reply:
I am not a network engineer so that it is difficult to say for me.
But
1. This commit fixs compilation error with keeping current network structure.
2. The recvmsg function just calls lwip API. It looks having strong dependancy with LWIP.
3. In POSIX standard, the recvmsg is in <sys/socket.h> like ours.
  If you want to support recvmsg, you must keep it here, not lwip's socket header.